marketplacetests/marketplace_gaia_test.py

```diff
@@ -29,32 +29,28 @@ def setUp(self):
         # This is used to tell FxA whether to create a dev or prod account
         self.base_url = 'https://marketplace-dev.allizom.org'
 
-    def install_in_app_payments_test_app(self):
+    def install_in_app_payments_test_app(self, app_name):
 
-        self.homescreen = Homescreen(self.marionette)
+        homescreen = Homescreen(self.marionette)
 
-        if not self.apps.is_app_installed(self.app_name):
+        # Remove the app if already installed
+        if self.apps.is_app_installed(app_name):
+            self.apps.uninstall(app_name)
```

Comment:
I'd personally prefer we had a tearDown where the app was removed and an assertion that the application is not already installed. If it's installed then I feel like it's a failure we should deal with. Perhaps this could be a later modification though.